vm/src/stdlib/collections.rs

```diff
@@ -0,0 +1,346 @@
+use crate::function::OptionalArg;
+use crate::obj::{objbool, objsequence, objtype::PyClassRef};
+use crate::pyobject::{IdProtocol, PyClassImpl, PyIterable, PyObjectRef, PyRef, PyResult, PyValue};
+use crate::vm::ReprGuard;
+use crate::VirtualMachine;
+use itertools::Itertools;
+use std::cell::{Cell, RefCell};
+use std::collections::VecDeque;
+
+#[pyclass(name = "deque")]
+#[derive(Debug, Clone)]
+struct PyDeque {
+    deque: RefCell<VecDeque<PyObjectRef>>,
+    maxlen: Cell<Option<usize>>,
+}
+
+impl PyValue for PyDeque {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("_collections", "deque")
```

Comment:
We really need a better solution for this construction in general. Maybe we could open an issue for it, asking other people for ideas on how to solve this?